crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4023,7 +4025,37 @@ mod tests {
         )?;
 
         // TODO should be a generic type
```

Comment:
We can remove this todo; this test is correct after this PR, there's nothing left to do here.